spec/lib/authorities/linked_data/generic_authority_spec.rb

```diff
@@ -69,50 +69,75 @@
         context "and lang NOT passed in" do
-          context "and NO default defined in config" do
-            let(:lod_lang_no_defaults) { described_class.new(:LOD_LANG_NO_DEFAULTS) }
-            let :results do
-              stub_request(:get, "http://localhost/test_no_default/search?query=milk")
-                .to_return(status: 200, body: webmock_fixture("lod_lang_search_enfr.rdf.xml"), headers: { 'Content-Type' => 'application/rdf+xml' })
-              lod_lang_no_defaults.search('milk')
+          context "and NO language defined in authority config" do
+            context "and NO language defined in Qa config" do
+              let(:lod_lang_no_defaults) { described_class.new(:LOD_LANG_NO_DEFAULTS) }
+              let :results do
+                stub_request(:get, "http://localhost/test_no_default/search?query=milk")
+                  .to_return(status: 200, body: webmock_fixture("lod_lang_search_enfr.rdf.xml"), headers: { 'Content-Type' => 'application/rdf+xml' })
+                lod_lang_no_defaults.search('milk')
+              end
+
+              before do
+                Qa.config.default_language = []
+              end
+
+              after do
+                Qa.config.default_language = :en
+              end
+
+              it "is not filtered" do
+                expect(results.first[:label]).to eq('[buttermilk, Babeurre] (yummy, délicieux)')
+                expect(results.second[:label]).to eq('[condensed milk, lait condensé] (creamy, crémeux)')
+                expect(results.third[:label]).to eq('[dried milk, lait en poudre] (powdery, poudreux)')
+              end
             end
-            it "is not filtered" do
-              expect(results.first[:label]).to eq('[buttermilk, Babeurre] (yummy, délicieux)')
-              expect(results.second[:label]).to eq('[condensed milk, lait condensé] (creamy, crémeux)')
-              expect(results.third[:label]).to eq('[dried milk, lait en poudre] (powdery, poudreux)')
+
+            context "and default_language is defined in Qa config" do
+              let(:lod_lang_no_defaults) { described_class.new(:LOD_LANG_NO_DEFAULTS) }
+              let :results do
+                stub_request(:get, "http://localhost/test_no_default/search?query=milk")
+                  .to_return(status: 200, body: webmock_fixture("lod_lang_search_enfr.rdf.xml"), headers: { 'Content-Type' => 'application/rdf+xml' })
+                lod_lang_no_defaults.search('milk')
+              end
+              it "filters using Qa configured default" do
+                expect(results.first[:label]).to eq('buttermilk (yummy)')
+                expect(results.second[:label]).to eq('condensed milk (creamy)')
+                expect(results.third[:label]).to eq('dried milk (powdery)')
+              end
             end
           end
 
-          context "and default IS defined in config" do
+          context "and language IS defined in authority config" do
```

Comment:
Changed some of the context labels to distinguish more clearly between the Qa default language and the language configured in the authority config file.